corehq/apps/performance_sms/forms.py

```diff
@@ -106,3 +95,80 @@ def source_id(self):
         if self.config.template_variables:
             return self.config.template_variables[0].source_id
         return ''
+
+
+def _get_default_form_helper():
+    helper = FormHelper()
+    helper.label_class = 'col-sm-3 col-md-2'
+    helper.field_class = 'col-sm-9 col-md-8 col-lg-6'
+    helper.form_class = "form-horizontal"
+    helper.form_id = "performance-form"
+    helper.form_method = 'post'
+    return helper
+
+
+def _clean_template(template):
+    try:
+        parser.validate(template)
+    except InvalidParameterException as e:
+        raise forms.ValidationError(unicode(e))
+    return template
+
+
+class AdvancedPerformanceMessageEditForm(forms.Form):
+    recipient_id = forms.CharField()
+    schedule = forms.ChoiceField(choices=[(choice, ugettext_lazy(choice)) for choice in SCHEDULE_CHOICES])
+    template = forms.CharField(widget=forms.Textarea)
+    template_variables = JsonField(expected_type=list)
+
+    def __init__(self, domain, config, *args, **kwargs):
+        self.domain = domain
+        self.config = config
+        super(AdvancedPerformanceMessageEditForm, self).__init__(initial=config.to_json(), *args, **kwargs)
+
+        self.fields['recipient_id'] = GroupField(domain=domain, label=_('Recipient Group'))
+
+        self.helper = _get_default_form_helper()
+        form_layout = self.fields.keys()
+        form_layout.append(
+            hqcrispy.FormActions(
+                StrictButton(
+                    _("Save Changes"),
+                    type="submit",
+                    css_class="btn btn-primary",
+                ),
+            )
+        )
+        self.helper.layout = Layout(
+            *form_layout
+        )
+
+    def clean_template(self):
+        return _clean_template(self.cleaned_data['template'])
+
+    def clean_schedule(self):
+        # todo: support other scheduling options
+        return ScheduleConfiguration(interval=self.cleaned_data['schedule'])
+
+    def save(self, commit=True):
+        self.config.recipient_id = self.cleaned_data['recipient_id']
+        self.config.schedule = self.cleaned_data['schedule']
+        self.config.template = self.cleaned_data['template']
+        self.config.template_variables = self.cleaned_data['template_variables']
+        if commit:
+            self.config.save()
+        return self.config
+
+    def clean_template_variables(self):
+        template_vars = self.cleaned_data['template_variables']
+        if not isinstance(template_vars, list):
+            raise forms.ValidationError(_('Template variables must be a list!'))
+        wrapped_vars = []
+        for var in template_vars:
+            try:
+                wrapped = TemplateVariable.wrap(var)
+                wrapped.validate()
+                wrapped_vars.append(wrapped)
```

Comment:
Should the `PerformanceMessageEditForm` also validate the `TemplateVariable` in this way?